src/client.rs

```diff
@@ -92,7 +92,13 @@ impl<'a, T: Authenticator + 'a> CDRS<T> {
             let body = start_response.get_body();
             let authenticator = body.get_authenticator().unwrap();
 
-            if authenticator.as_str() == self.authenticator.get_cassandra_name() {
+            if self.authenticator.get_cassandra_name() == "NONE" {
+                let io_err = io::Error::new(
+                    io::ErrorKind::NotFound,
+                    format!("No authenticator was provided "));
+                return Err(error::Error::Io(io_err));
+
+            }else if authenticator.as_str() == self.authenticator.get_cassandra_name().to_string() {
```

Comment:
> Maybe we should just use Option because AuthenticationNone which returns None feels over complicated

that would mean changing the trait as well

```
pub trait Authenticator: Clone {
    fn get_auth_token(&self) -> CBytes;
    fn get_cassandra_name(&self) -> &str;
}

```
to
```

pub trait Authenticator: Clone {
    fn get_auth_token(&self) -> CBytes;
    fn get_cassandra_name(&self) -> Option<&str>;
}
```
+ refactoring rest of code; I was trying to sneak in minimum code changes 
what do you feel ?
or may be an Enum ?